### PR TITLE
prismaConverterDev

### PR DESCRIPTION
**Pull request by georgedias**

### CODE REVIEW

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      target_id: 'Prisma Cloud Scan Report'
+    },
+    version: HeimdallToolsVersion,
+    statistics: {
+      duration: null
+    },
+    profiles: [
+      {
+        name: 'Palo Alto Prisma Cloud Tool',
+        version: HeimdallToolsVersion,
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:49]: this should be specifying the version of the profile/test/prisma/whatever, not heimdall tools.  it also should not be "package version".  if that information is not provided, leave it as the empty string.

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                    } else if (obj.Type === 'linux') {
+                      if (obj.Distro !== '') {
+                        result += `Configuration check for ${obj.Distro}`;
+                      } else {
+                        result += ``;
+                      }
+                    } else {
+                      result += `${obj.Type} check for ${obj.Hostname}`;
+                    }
+                    result += `\n\n${obj.Description}`;
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:120]: imo better to have the description in the control's desc tag

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        attributes: [],
+        depends: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'records',
+            key: 'id',
+            tags: {
+              cve: {path: 'CVE ID'},
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:66]: would probably add a cve published date tag and put the Published column in there

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+              cvss: {path: 'cssv'}
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:67]: the header is wrong - it should be all caps i think

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                }
+              }
+            },
+            code: {
+              transformer: (obj: Record<string, string>) =>
+                JSON.stringify(obj, null, 2)
+            },
+            results: [
+              {
+                status: ExecJSON.ControlResultStatus.Failed,
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:97]: will there only ever be fails?

https://docs.paloaltonetworks.com/prisma/prisma-cloud/prisma-cloud-admin-compute/compliance/compliance_explorer.html

makes it seem like there might be passes as well

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+        status: 'loaded',
+        controls: [
+          {
+            path: 'records',
+            key: 'id',
+            tags: {
+              cve: {path: 'CVE ID'},
+              cvss: {path: 'cssv'}
+            },
+            descriptions: [],
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:69]: might need to do conditional processing of the 'Description' in order to extract stuff like 'Mitigation' (analogous to fix text) and put it in here.  i think @ejaronne would be pretty interested in going through the possible variations that can exist within this field

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+        supports: [],
+        attributes: [],
+        depends: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'records',
+            key: 'id',
+            tags: {
```
Comment (Amndeep7) [inline libs/hdf-converters/src/prisma-mapper.ts:65]: i think you're missing a bunch of headers that you can probably shove in here or just leave in the code block i guess.  these include things like service/cluster/source package/etc.

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

**Comment by Amndeep7** [comment]
this is just stuff that i saw going through it relatively quickly.  i think @ejaronne should probably also go over this to figure out how best to map things.  that Description header in particular i think is gonna be a lot more work.

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+  'Fix Status'?: string;
+  Cause?: string;
+};
+
+const SEVERITY_LOOKUP: Record<string, number> = {
+  low: 0.3,
+  moderate: 0.5,
+  high: 0.7,
+  important: 0.9,
+  critial: 1
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:30]:
```suggestion
  critical: 1
```

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+};
+
+const DEFAULT_NIST_TAG = ['SA-11', 'RA-5'];
+const BASE_NIST_TAG = ['SI-2', 'RA-5'];
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:34]: Please differentiate between these two (either in a comment or by variable name)

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+function getId(value: Record<string, string>): string {
+  const complianceID = _.get(value, 'Compliance ID');
+  const cveID = _.get(value, 'CVE ID');
+  const distro = _.get(value, 'Distro');
+  const severity = _.get(value, 'Severity');
+  return complianceID.concat(
+    DASH,
+    cveID ? cveID : `${distro}${DASH}${severity}`
+  );
+}
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:53]:
```suggestion
function nistTag(cveTag: Record<string, string>): string[] {
  if (!cveTag) {
    return DEFAULT_NIST_TAG;
  } else {
    return BASE_NIST_TAG;
  }
}
function getId(value: Record<string, string>): string {
  const complianceID = _.get(value, 'Compliance ID');
  const cveID = _.get(value, 'CVE ID');
  const distro = _.get(value, 'Distro');
  const severity = _.get(value, 'Severity');
  if (cveID) {
    return `${complianceID}-${cveID}`;
  } else {
    return `${complianceID}-${distro}-${severity}`;
  }
}
```

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+  critical: 1
+};
+
+// The DEFAULT_NIST_TAG is applicable to the following controls:
+// SA-11 (DEVELOPER SECURITY TESTING AND EVALUATION) - RA-5 (VULNERABILITY SCANNING)
+const DEFAULT_NIST_TAG = ['SA-11', 'RA-5'];
+
+// The REMEDIATION_NIST_TAG is applicable to the following controls:
+// SI-2 (FLAW REMEDIATION) - 	RA-5 (VULNERABILITY SCANNING)
+const REMEDIATION_NIST_TAG = ['SI-2', 'RA-5'];
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:39]:
```suggestion
// The REMEDIATION_NIST_TAG the set of default applicable NIST 800-53 controls for ensuring up-to-date packages.
// SI-2 (FLAW REMEDIATION) - 	RA-5 (VULNERABILITY SCANNING)
const REMEDIATION_NIST_TAG = ['SI-2', 'RA-5'];
```

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+  low: 0.3,
+  moderate: 0.5,
+  high: 0.7,
+  important: 0.9,
+  critical: 1
+};
+
+// The DEFAULT_NIST_TAG is applicable to the following controls:
+// SA-11 (DEVELOPER SECURITY TESTING AND EVALUATION) - RA-5 (VULNERABILITY SCANNING)
+const DEFAULT_NIST_TAG = ['SA-11', 'RA-5'];
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:35]:
```suggestion
// The DEFAULT_NIST_TAG is applicable to all automated configuration tests.
// SA-11 (DEVELOPER SECURITY TESTING AND EVALUATION) - RA-5 (VULNERABILITY SCANNING)
const DEFAULT_NIST_TAG = ['SA-11', 'RA-5'];
```

File: libs/hdf-converters/src/prisma-mapper.ts [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+    return `${complianceID}-${cveID}`;
+  } else {
+    return `${complianceID}-${distro}-${severity}`;
+  }
+}
+function getTitle(item: Record<string, string>): string {
+  const hostName = _.get(item, 'Hostname');
+  const distro = _.get(item, 'Distro');
+  const type = _.get(item, 'Type');
+  return hostName.toString().concat(distro, type);
```
Comment (camdenmoors) [inline libs/hdf-converters/src/prisma-mapper.ts:63]:
```suggestion
  return `${hostName}-${distro}-${type}`;
```

**Review by camdenmoors — APPROVED** [review]

**Pull request merged**